scripts/calendar/hebrew_generator.py

```
@@ -44,49 +44,134 @@
+            return 2
+
+        last_year_len = self.elapsed_days(year - 1)
+        if present_year_len - last_year_len == 382:
+            return 1
+
+        return 0
+
+    def new_year(self, year: int) -> int:
+        return HEBREW_EPOCH_RD + self.elapsed_days(year) + self.year_length_correction(year)
+
+    @cache
```

Ruff (0.15.9)

[warning] Use of `functools.lru_cache` or `functools.cache` on methods can lead to memory leaks

(B019)